### PR TITLE
Feat/89 implement cancel subscription confirmation prompt modal

### LINKED ISSUE #89: [FEAT] - SETTINGS PAGE -> Implement Cancel Subscription Confirmation Prompt Modal
### **Description:**
Implement a modal that prompts users to confirm their decision to cancel their subscription. The modal should provide two options: Keep Subscription and Cancel Subscription. The Keep Subscription button will close the modal, allowing users to continue with their current subscription, while the Cancel Subscription button will proceed with canceling the user's subscription.

### **Acceptance Criteria:**
1. The modal should be triggered when users choose to cancel their subscription.
2. The modal should contain a clear message asking for confirmation to cancel the subscription.
3. There should be two buttons: Keep Subscription and Cancel Subscription.
4. The Keep Subscription button should close the modal without making any changes.
5. The Cancel Subscription button should cancel the user's subscription and close the modal.
6. The modal should be responsive and accessible.
7. Appropriate success or error messages should be displayed based on the outcome.
8. Unit tests should cover the modal's functionality.
9. Integration tests should ensure proper interaction with the backend for subscription cancellation.

### **Expected Outcome:**
A functional confirmation prompt modal that provides users with a clear choice to either keep their subscription or proceed with canceling it. The modal should be user-friendly, responsive, and accessible, ensuring a smooth user experience.

### **Purpose:**
The confirmation prompt modal will provide a final safeguard against accidental subscription cancellations, ensuring that users consciously confirm their decision.

### **Requirement:**
1. Design and implement the confirmation prompt modal as per the provided Figma designs.
2. Ensure the modal contains a clear confirmation message.
3. Implement two buttons: Keep Subscription and Cancel Subscription.
4. Integrate with the backend API to handle subscription cancellation.
5. Ensure the modal is responsive across all devices.
6. Implement error handling and display appropriate messages.
7. Follow accessibility guidelines for modal implementation.

### **Testing:**
1. Unit tests for the modal's functionality.
2. Integration tests to ensure proper interaction with the backend for subscription cancellation.
3. Manual testing across different devices to ensure responsiveness.

### **Task:**
- [ ] Set up the confirmation prompt modal component.
- [ ] Add a clear confirmation message to the modal.
- [ ] Implement the Keep Subscription button to close the modal.
- [ ] Implement the Cancel Subscription button to cancel the user's subscription and close the modal.
- [ ] Integrate with the backend API for subscription cancellation.
- [ ] Ensure the modal is responsive across all devices.
- [ ] Add error handling and display appropriate messages.
- [ ] Write unit tests for the modal's functionality.
- [ ] Conduct integration tests with the backend.
- [ ] Perform manual testing for responsiveness and accessibility.

### **Picture of the corresponding component/page of the ticket from the Figma:**

Dialog card

### **Figma Link:**
- [Figma](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=433-48633&m=dev)

### PR DESCRIPTION
**Pull request by VeraVechy**
# Description

**Closes #89 **

# Changes proposed

## What did you do?

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

**Comment by VeraVechy** [comment]
@oreoluwa212 please find fya. Regards.

File: src/app/about/button.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import * as React from "react"
```
Comment (kleenpulse) [inline src/app/about/button.tsx]: A button component has already been provided, you should navigate to localhost:3000/guides for more info

File: src/app/guides/page.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import React from "react";
 import CustomButton from "~/components/common/Button/button";
 import Sidebar from "~/components/layouts/Sidebar";
 
-const Button: React.FC = () => {
+const Btn: React.FC = () => {
```
Comment (kleenpulse) [inline src/app/guides/page.tsx]: Import and use the already provided button from style guide

File: src/app/about/page.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+"use client";
+
+import React, { useState } from "react";
+
+import ConfirmCancelModal from "../../components/cancel-sub-modal/ConfirmCancelModal";
+import { Button } from "../about/button";
+
+const AboutPage = () => {
```
Comment (kleenpulse) [inline src/app/about/page.tsx]: this is suppose to be component not a page, please refer to the git issue and read through it carefully, understand the task, tick each checkbox as the you do them and lastly attach a screenshot or a recorded video.
Run tests, if they all pass, take a screenshot and upload it as well.

**Review by kleenpulse — CHANGES_REQUESTED** [review]
Ensure you do the task correctly, seek help if needed from your fellow interns, otherwise your task would be unassigned from you and assign to someone else

**Comment by VeraVechy** [comment]
Please find attached, screenshots of tests conducted. The component I worked on passed all tests, that is the "cancel-sub-modal' folder.

Dear reviewer, in case this PR is still not satisfactory, I kindly request that in order not to delay the team any further, this component could be reassigned to another person. 

Regards.

**Pull request closed without merging**